shell/layouts/default.vue

```diff
@@ -194,224 +209,101 @@ export default {
-      }
-    },
-
-    clusterId(a, b) {
-      if ( !isEqual(a, b) ) {
-        // Store the last visited route when the cluster changes
-        this.setClusterAsLastRoute();
-      }
-    },
-
-    namespaceMode(a, b) {
-      if ( !isEqual(a, b) ) {
-        // Immediately update because you'll see it come in later
-        this.getGroups();
-      }
-    },
-
-    namespaces(a, b) {
-      if ( !isEqual(a, b) ) {
-        // Immediately update because you'll see it come in later
-        this.getGroups();
-      }
-    },
-
-    clusterReady(a, b) {
-      if ( !isEqual(a, b) ) {
-        // Immediately update because you'll see it come in later
-        this.getGroups();
-      }
-    },
-
-    product(a, b) {
-      if ( !isEqual(a, b) ) {
-        // Immediately update because you'll see it come in later
-        this.getGroups();
-      }
-    },
-
-    async currentProduct(a, b) {
-      if ( !isEqual(a, b) ) {
-        if (a.inStore !== b.inStore || a.inStore !== 'cluster' ) {
-          const route = {
-            name:   'c-cluster-product',
-            params: {
-              cluster: this.clusterId,
-              product: a.name,
-            }
-          };
+    loadedProducts() {
+      const loadProducts = this.isExplorer ? [EXPLORER] : [];
+      const currentProduct = this.productId; // Use the productId computed property as a dependency to trigger updates when the product changes.
 
-          await this.$store.dispatch('prefs/setLastVisited', route);
+      if (this.isExplorer) {
+        for (const product of this.activeProducts) {
+          if (product.inStore === 'cluster') {
+            addObject(loadProducts, product.name);
+          }
         }
       }
-    },
-
-    $route(a, b) {
-      this.$nextTick(() => this.syncNav());
-    },
-
-  },
-
-  async created() {
-    this.queueUpdate = debounce(this.getGroups, 500);
-
-    this.getGroups();
-
-    await this.$store.dispatch('prefs/setLastVisited', this.$route);
-  },
-
-  mounted() {
-    // Sync the navigation tree on fresh load
-    this.$nextTick(() => this.syncNav());
-  },
-
-  methods: {
-    async setClusterAsLastRoute() {
-      const route = {
-        name:   this.$route.name,
-        params: {
-          ...this.$route.params,
-          cluster: this.clusterId,
-        }
-      };
 
-      await this.$store.dispatch('prefs/setLastVisited', route);
-    },
-    handlePageAction(action) {
-      if (action.action === SET_LOGIN_ACTION) {
-        this.afterLoginRoute = this.getLoginRoute();
-        // Mark release notes as seen, so that the login route is honoured
-        markSeenReleaseNotes(this.$store);
-      }
-    },
+      // This should already have come into the list from above, but in case it hasn't...
+      addObject(loadProducts, currentProduct);
 
-    getLoginRoute() {
-      return {
-        name:   this.$route.name,
-        params: this.$route.params
-      };
+      return loadProducts;
     },
 
-    collapseAll() {
```

Comment:
For now, I left the manual updates of `collapseAll` and the nav sync method, though they could be refactored in a similar way if we wanted to.